auth-agents-common/src/main/java/org/apache/atlas/policytransformer/PurposeCachePolicyTransformer.java

```diff
@@ -0,0 +1,121 @@
+        purposeTemplate = getTemplate(TEMPLATE_SUFFIX);
+        this.entityRetriever = entityRetriever;
+    }
+
+    public List<AtlasEntityHeader> transform(AtlasEntityHeader atlasPolicy) {
+        AtlasPerfMetrics.MetricRecorder recorder = RequestContext.get().startMetricRecord("PurposeCachePolicyTransformer.transform");
+        List<AtlasEntityHeader> ret = new ArrayList<>();
+
+        List<String> atlasActions = getPolicyActions(atlasPolicy);
+        List<String> atlasResources = getPolicyResources(atlasPolicy);
+        List<String> tags = getFilteredPolicyResources(atlasResources, RESOURCES_TAG);
+
+        int index = 0;
+        Set<String> templateActions = purposeTemplate.getTemplateActions();
+        List<String> transformableActions = (List<String>) CollectionUtils.intersection(atlasActions, templateActions);
+
+
+        for (String atlasAction : transformableActions) {
+            List<PolicyTransformerTemplate.TemplatePolicy> currentTemplates = purposeTemplate.getTemplate(atlasAction);
+
+            if (CollectionUtils.isEmpty(currentTemplates)) {
+                LOG.warn("PurposeCachePolicyTransformer: Skipping unknown action {} while transforming policy {}", atlasAction, atlasPolicy.getGuid());
+                continue;
+            }
+
+            for (PolicyTransformerTemplate.TemplatePolicy templatePolicy : currentTemplates) {
+                AtlasEntityHeader header = new AtlasEntityHeader(atlasPolicy);
+
+                header.setGuid(atlasPolicy.getGuid() + "-" + index++);
+
+                header.setAttribute(ATTR_POLICY_ACTIONS, templatePolicy.getActions());
+                header.setAttribute(ATTR_POLICY_RESOURCES_CATEGORY, templatePolicy.getPolicyResourceCategory());
+                header.setAttribute(ATTR_POLICY_IS_ENABLED, getIsPolicyEnabled(atlasPolicy));
+
+                if (StringUtils.isNotEmpty(templatePolicy.getPolicyServiceName())) {
+                    header.setAttribute(ATTR_POLICY_SERVICE_NAME, templatePolicy.getPolicyServiceName());
+                }
+
+                List<String> finalResources = new ArrayList<>();
+
+                for (String templateResource : templatePolicy.getResources()) {
+                    if (templateResource.contains(PLACEHOLDER_TAG)) {
+                        tags.forEach(tag -> finalResources.add(templateResource.replace(PLACEHOLDER_TAG, tag)));
+                    } else {
+                        finalResources.add(templateResource);
+                    }
+                }
+                header.setAttribute(ATTR_POLICY_RESOURCES, finalResources);
+
+                ret.add(header);
+            }
+        }
+
+        //prepare a policy for all non-transformable actions
+        //this will help to reduce number of overall transformed policies
+        List<String> nonTransformableActions = (List<String>) CollectionUtils.subtract(atlasActions, templateActions);
+
+        if (CollectionUtils.isNotEmpty(nonTransformableActions)) {
```

Comment:
What could be these policies?

Reply:
it will be separate tag-based policy with all remaining actions, I will post an example here

Reply:
Assume creating a purpose metadata policy with following actions:
"entity-create",
"entity-delete",
"entity-read",
"entity-remove-classification",
"entity-add-classification",
"entity-update-classification",
"entity-update-business-metadata",
"purpose-add-terms",
"purpose-remove-terms",
"purpose-entity-update"

Now there will be 
1. Resource based policy to support purpose-add-terms
2. Resource based policy to support purpose-remove-terms
3. Resource based policy to support add/remove Links, Readme for purpose-entity-update
4. Tag based policy to support entity-update for purpose-entity-update
5. Tag based policy to support remaining actions like
> - "entity-create",
> - "entity-delete",
> - "entity-read",
> - "entity-remove-classification",
> - "entity-add-classification",
> - "entity-update-classification",
> - "entity-update-business-metadata"

This section handles 5th pointer